src/main/kotlin/org/rust/ide/statistics/RsExternalLinterUsagesCollector.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.statistics
+
+import com.intellij.internal.statistic.eventLog.EventLogGroup
+import com.intellij.internal.statistic.eventLog.events.EventFields.Long
+import com.intellij.internal.statistic.service.fus.collectors.CounterUsagesCollector
+
+@Suppress("UnstableApiUsage")
+class RsExternalLinterUsagesCollector : CounterUsagesCollector() {
+
+    override fun getGroup(): EventLogGroup = GROUP
+
+    companion object {
+        private val GROUP = EventLogGroup("rust.external.linter", 1)
+
+        private val ON_THE_FLY_EXECUTION_TIME = GROUP.registerEvent("on.the.fly.execution", Long("duration_ms"))
```

Comment:
By reporting exact `Long` values we won't be ably to query distribution of performance efficiently since (f.i. to know what percentage of requests are less than 1s). Consider using `RoundedLong` or event better - concrete values we want to check agains.